lib/kubeclient.rb

```diff
@@ -138,5 +129,51 @@ def all_entities
         result_hash[key_name] = send(method_name)
       end
     end
+
+    def watch(entities)
+      watch_endpoint = api_endpoint + '/watch/'
+      options = { inactivity_timeout: 0, keepalive: true }
+
+      EventMachine.run do
+        multi_request = EventMachine::MultiRequest.new
+
+        entities.each do |entity, version|
+          uri = URI.parse(watch_endpoint + get_resource_name(entity.to_s))
+
+          unless version.nil?
+            uri.query = URI.encode_www_form('resourceVersion' => version)
+          end
+
+          request = EventMachine::HttpRequest.new(uri, options).get
+
+          request.errback do
+            EventMachine.stop
+            fail request.error
+          end
+
+          request.callback do
+            EventMachine.stop
+            unless request.response_header.successful?
+              fail KubeException.new(
+                request.response_header.status,
+                request.response_header.http_reason
+              )
+            end
+          end
+
+          buffer = ''
+          request.stream do |chunk|
+            if request.response_header.successful?
+              buffer += chunk
+              while (notice = buffer.slice!(/.+\n/))
```

Comment:
If multiple messages come through,  this might miss one,  I think, since the regex is greedy